sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -908,6 +915,215 @@ class RotatingCertificateCredential implements TokenCredential {
+
+```ts
+import { useIdentityExtension } from "@azure/identity";
+import { cachePersistenceExtension } from "@azure/identity-cache-persistence";
+
+useIdentityExtension(cachePersistenceExtension);
+```
+
+After calling `useIdentityExtension`, the persistent token cache extension is registered to the `@azure/identity` package and will be available on all credentials that support persistent token caching (those that have `tokenCachePersistenceOptions` in their constructor options).
+
+#### Persisting the token cache
+
+The credential handles persisting all the data needed to silently authenticate one or many accounts. It manages sensitive data such as refresh tokens and access tokens which must be protected to prevent compromising the accounts related to them. By default, the `@azure/identity` library will protect and cache sensitive token data using available platform data protection.
+
+To configure a credential, such as the `InteractiveBrowserCredential`, to persist token data, simply set the `tokenCachePersistenceOptions` option.
+
+```ts
+import { useIdentityExtension, InteractiveBrowserCredential } from "@azure/identity";
+import { cachePersistenceExtension } from "@azure/identity-cache-persistence";
+
+useIdentityExtension(cachePersistenceExtension);
+
+const credential = new InteractiveBrowserCredential({
+  tokenCachePersistenceOptions: {
+    enabled: true
+  }
+});
+```
+
+#### Persisting the Authentication Record
+
+The `AuthenticationRecord` which is returned from the `authenticate`, contains data identifying an authenticated account. It is needed to identify the appropriate entry in the persisted token cache to silently authenticate on subsequent executions. There is no sensitive data in the `AuthenticationRecord` so it can be persisted in a non-protected state. Ensure that you pass in the appropriate scopes for your service to `authenticate` method.
+
+Here is an example of an application storing the `AuthenticationRecord` to the local file system after authenticating the user.
+
+```ts
+import {
+  useIdentityExtension,
+  InteractiveBrowserCredential,
+  AuthenticationRecord,
+  serializeAuthenticationRecord
+} from "@azure/identity";
+import { cachePersistenceExtension } from "@azure/identity-cache-persistence";
+import fs from "fs";
+import { promisify } from "util";
+import path from "path";
+
+useIdentityExtension(cachePersistenceExtension);
+
+export async function main(): Promise<void> {
+  const AUTH_RECORD_PATH = "./tokencache.bin";
+  const credential = new InteractiveBrowserCredential({
+    tokenCachePersistenceOptions: {
+      enabled: true
+    }
+  });
+  const authRecord: AuthenticationRecord = await credential.authenticate(
+    "https://service/.default"
+  );
+  const writeFileAsync = promisify(fs.writeFile);
```

Comment:
I wouldn't use util.promisify here, as it's not a core part of the example. It adds extra concepts that aren't relevant to the identity features you're demonstrating, so I would just use `fs.writeFileSync`.

Reply:
updated